### ISSUE TITLE
Installing pylint causes ward to fail with "No module named 'pytest'"

### ISSUE THREAD

**Issue by anlutro** [darrenburns/ward#157]
I have a project where I'm trying out ward as the test runner. It's working fine, until I also install pylint, at which point trying to run `ward` yields this traceback:

```
$ ward
Traceback (most recent call last):
  File "/home/andreas/code/systemd-multi-process-manager/.venv/bin/ward", line 8, in <module>
    sys.exit(run())
  File "/home/andreas/code/systemd-multi-process-manager/.venv/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/andreas/code/systemd-multi-process-manager/.venv/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/andreas/code/systemd-multi-process-manager/.venv/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/andreas/code/systemd-multi-process-manager/.venv/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/andreas/code/systemd-multi-process-manager/.venv/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/andreas/code/systemd-multi-process-manager/.venv/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/andreas/code/systemd-multi-process-manager/.venv/lib/python3.8/site-packages/ward/run.py", line 134, in test
    modules = list(load_modules(mod_infos))
  File "/home/andreas/code/systemd-multi-process-manager/.venv/lib/python3.8/site-packages/ward/collect.py", line 109, in load_modules
    m.__loader__.exec_module(m)
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File ".venv/lib/python3.8/site-packages/astroid/test_utils.py", line 17, in <module>
    import pytest
ModuleNotFoundError: No module named 'pytest'
```

**Comment by anlutro** [comment]
Added a `print(m)` right before the line in `collect.py` which causes the import:

```
$ ward
<module 'test_sdmpm' from './test_sdmpm.py'>
<module 'test_utils' from '.venv/lib/python3.8/site-packages/astroid/test_utils.py'>
```

So it seems a file from my virtualenv is being picked up because it matches the `test_*` pattern.

**Comment by darrenburns** [comment]
Thanks for the report! This is a known issue that's been open for a while, but nobody has been able to pick it up yet: https://github.com/darrenburns/ward/issues/81

**Comment by darrenburns** [comment]
Fixed in Ward 0.44.0b0, which is now available on PyPI.
